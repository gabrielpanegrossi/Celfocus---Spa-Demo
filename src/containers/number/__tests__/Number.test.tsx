import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter as Router } from 'react-router-dom';
import * as MockReactRouterDom from 'react-router-dom';
import Number from '../index';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual<typeof MockReactRouterDom>('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

jest.mock('react-query', () => ({
  useQuery: () => ({
    isLoading: false,
    error: undefined,
    data: { id: '351910000000', type: 'mobile', company_id: 1 },
  }),
}));

describe('ContainerHeader', () => {
  beforeEach(() => {
    render(
      <Router>
        <Number />
      </Router>
    );
  });

  it('should match snpashot', () => {
    const container = render(
      <Router>
        <Number />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render', () => {
    expect(screen.queryByRole('button', { name: 'Back' })).toBeInTheDocument();
  });

  it('should render fetched info on table', async () => {
    expect(screen.queryByRole('table')).toBeInTheDocument();

    await waitFor(() => {
      expect(screen.queryByText('351 910000000')).toBeInTheDocument();
      expect(screen.queryByText('mobile')).toBeInTheDocument();
    });
  });

  it('should navigate back', async () => {
    const user = userEvent.setup();

    user.click(screen.queryByRole('button', { name: 'Back' }));

    await waitFor(() => {
      expect(mockNavigate).toBeCalledWith(-1);
    });
  });
});
