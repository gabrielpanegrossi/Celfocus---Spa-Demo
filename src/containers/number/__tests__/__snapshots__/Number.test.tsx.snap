// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerHeader should match snpashot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-crXcEl kSbmp"
      >
        <header
          class="sc-bczRLJ bXXmoA"
        >
          <div
            class="sc-gsnTZi jGtclP"
          >
            <button
              class="sc-dkzDqf dUlZGR"
            >
              Back
            </button>
          </div>
          <div
            class="sc-gsnTZi jGtclP"
          >
            <h1
              class="sc-hKMtZM cfLWee"
            />
          </div>
          <div
            class="sc-gsnTZi jGtclP"
          />
        </header>
        <div
          class="sc-evZas gXkgsR"
        >
          <table
            class="sc-eCYdqJ vSbeb"
          >
            <thead>
              <tr>
                <th>
                  Number
                </th>
                <th>
                  Type
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  351 910000000
                </td>
                <td>
                  mobile
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div>
      <div
        class="sc-crXcEl kSbmp"
      >
        <header
          class="sc-bczRLJ bXXmoA"
        >
          <div
            class="sc-gsnTZi jGtclP"
          >
            <button
              class="sc-dkzDqf dUlZGR"
            >
              Back
            </button>
          </div>
          <div
            class="sc-gsnTZi jGtclP"
          >
            <h1
              class="sc-hKMtZM cfLWee"
            />
          </div>
          <div
            class="sc-gsnTZi jGtclP"
          />
        </header>
        <div
          class="sc-evZas gXkgsR"
        >
          <table
            class="sc-eCYdqJ vSbeb"
          >
            <thead>
              <tr>
                <th>
                  Number
                </th>
                <th>
                  Type
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  351 910000000
                </td>
                <td>
                  mobile
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-crXcEl kSbmp"
    >
      <header
        class="sc-bczRLJ bXXmoA"
      >
        <div
          class="sc-gsnTZi jGtclP"
        >
          <button
            class="sc-dkzDqf dUlZGR"
          >
            Back
          </button>
        </div>
        <div
          class="sc-gsnTZi jGtclP"
        >
          <h1
            class="sc-hKMtZM cfLWee"
          />
        </div>
        <div
          class="sc-gsnTZi jGtclP"
        />
      </header>
      <div
        class="sc-evZas gXkgsR"
      >
        <table
          class="sc-eCYdqJ vSbeb"
        >
          <thead>
            <tr>
              <th>
                Number
              </th>
              <th>
                Type
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                351 910000000
              </td>
              <td>
                mobile
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
